src/unittest/test_areastore.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+Minetest
+Copyright (C) 2015 est31, <MTest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "test.h"
+
+#include "areastore.h"
+
+class TestAreaStore : public TestBase {
+public:
+	TestAreaStore() { TestManager::registerTestModule(this); }
+	const char *getName() { return "TestAreaStore"; }
+
+	void runTests(IGameDef *gamedef);
+
+	void genericStoreTest(AreaStore *store);
+	void testVectorStore();
+	void testSpatialStore();
+};
+
+static TestAreaStore g_test_instance;
+
+void TestAreaStore::runTests(IGameDef *gamedef)
+{
+	TEST(testVectorStore);
+#if USE_SPATIAL
+	TEST(testSpatialStore);
+#endif
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+void TestAreaStore::testVectorStore()
+{
+	VectorAreaStore store;
+	genericStoreTest(&store);
+}
+
+void TestAreaStore::testSpatialStore()
+{
+#if USE_SPATIAL
+	SpatialAreaStore store;
+	genericStoreTest(&store);
+#endif
+}
+
+void TestAreaStore::genericStoreTest(AreaStore *store)
+{
+	Area a(v3s16(-10, -3, 5), v3s16(0, 29, 7));
+	a.id = 1;
+	Area b(v3s16(-5, -2, 5), v3s16(0, 28, 6));
+	b.id = 2;
+	Area c(v3s16(-7, -3, 6), v3s16(-1, 27, 7));
+	c.id = 3;
+	std::vector<Area *> res;
+
+	UASSERT(store->size() == 0);
```

Comment:
When comparing primitive types such as ints or even std::strings, try using UASSERT_EQ() so that the expected value and actual value are printed out as well which is more helpful than simply saying they aren't equal.